merkle/merkle_tree.go

```diff
@@ -0,0 +1,196 @@
+/*
+Package merkle provides functions and types to deal with the creation and storage of
+Merkle trees, using the secure SHAKE256 KDF implemented in the signing/key package:
+thus not being affected by the the infamous M-Bug.
+
+The functions exported by the package are:
+
+- CreateMerkleTree(seed trinary.Hash, securityLvl int, depth int) creates a MerkleTree
+  structure of the specified depth, using a SHAKE256 key of the the length specified by
+  the supplied securitylevel, deriving subseeds from the provided seed.
+- StoreMerkleTreeFile(filePath string, merkleTree *MerkleTree) stores the MerkleTree
+  structure in a file; the format used is compatible with Hornet.
+- LoadMerkleTreeFile(filePath string) loads a Hornet-compatible Merkle tree file as a
+  MerkleTree structure.
+*/
+package merkle
+
+import (
+	"math"
+	"runtime"
+	"sync"
+
+	"github.com/iotaledger/iota.go/consts"
+	"github.com/iotaledger/iota.go/kerl"
+	"github.com/iotaledger/iota.go/signing"
+	"github.com/iotaledger/iota.go/signing/key"
+	"github.com/iotaledger/iota.go/trinary"
+)
+
+// MerkleTree contains the merkle tree used for the coordinator signatures.
+type MerkleTree struct {
+	// depth of the merkle tree
+	Depth int
+	// root address of the merkle tree
+	Root trinary.Hash
+	// merkle tree layers indexed by their level
+	Layers map[int]*MerkleTreeLayer
+}
+
+// MerkleTreeLayer contains the nodes of a layer of a merkle tree.
+type MerkleTreeLayer struct {
+	// level of the layer in the tree
+	Level int
+	// nodes of the layer
+	Hashes []trinary.Hash
+}
+
+type MilestoneIndex uint32
```

Comment:
I think for the same reason why Trytes is a type instead of simple string. It better determines what object we are dealing with.
You are right about it not being used outside of this packge: I've unexported it.